packages/nvidia_nat_core/src/nat/authentication/credential_validator/bearer_token_validator.py

```
@@ -126,20 +130,54 @@ async def verify(self, token: str) -> TokenValidationResult:
             return TokenValidationResult(client_id="", token_type="bearer", active=False)
 
-        try:
-            if token.count(".") == 2:
+        can_introspect = bool(self.introspection_endpoint and self.client_id and self.client_secret)
+        if token.count(".") == 2:
+            try:
                 return await self._verify_jwt_token(token)
-            elif (self.introspection_endpoint and self.client_id and self.client_secret):
-                return await self._verify_opaque_token(token)
-            else:
+            except Exception as jwt_error:
```

Ruff (0.15.15)

[warning] Do not catch blind exception: `Exception`

(BLE001)